src/main/java/com/google/sps/data/UserStorage.java

```diff
@@ -76,6 +76,13 @@ public String getIDbyUsername(String username) {
     return userEntity != null ? (String) userEntity.getProperty("id") : null;
   }
 
+  public String getUsernameByID(String userId) {
+    Query query = new Query("User").setFilter(
+        new FilterPredicate(Entity.KEY_RESERVED_PROPERTY, FilterOperator.EQUAL, KeyFactory.createKey("User", userId)));
+    Entity userEntity = DatastoreServiceFactory.getDatastoreService().prepare(query).asSingleEntity();
+    return userEntity != null ? (String) userEntity.getProperty("email") : null;
```

Comment:
Do it like: `if (userEntity.getProperty("username")==null) return email else return username`